### PR TITLE
Mention configuration-cache types moving to internal package in the upgrade guide

### PR DESCRIPTION
**Pull request by abstratt**
Fixes: #29351

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 1]
```diff
@@ -59,6 +59,18 @@ This means it was possible for some projects to compile without any declared dep
 We believe this situation is very unlikely to occur in real projects, as IDE integration and test execution would not work correctly.
 If you need to use the Gradle API, declare a `gradleApi` dependency or use the `java-gradle-plugin` plugin.
 
+==== Configuration cache implementation packages now under `org.gradle.internal`
```
Comment (abstratt) [inline platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:62]: @bamboo I removed any references to specific packages and am stating all packages are gone

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 2]
```diff
@@ -59,6 +59,18 @@ This means it was possible for some projects to compile without any declared dep
...
+
+References to Gradle types that are not API should be avoided at all costs, as their direct use is definitely unsupported,
```
Comment (lkasso) [inline platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:64]:
```suggestion
References to Gradle types that are not part of the public API should be avoided, as their direct use is unsupported,
```

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 3]
```diff
@@ -59,6 +59,18 @@ This means it was possible for some projects to compile without any declared dep
...
+and Gradle internal implementation classes may suffer breaking changes (or be renamed or removed) from a version to another without any warning.
```
Comment (lkasso) [inline platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:65]:
```suggestion
and Gradle internal implementation classes may suffer breaking changes (or be renamed or removed) from one version to another without warning.
```

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 4]
```diff
@@ -59,6 +59,18 @@ This means it was possible for some projects to compile without any declared dep
...
+
+It is important that users are able to tell which parts of the Gradle codebase are API, and which parts are internal.
```
Comment (lkasso) [inline platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:67]:
```suggestion
It is important for users to distinguish between the API and internal parts of the Gradle codebase.
```

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 5]
```diff
@@ -59,6 +59,18 @@ This means it was possible for some projects to compile without any declared dep
...
+One way that is achieved is by having `internal` in implementation package names.
```
Comment (lkasso) [inline platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:68]:
```suggestion
This is typically achieved by including `internal` in the implementation package names.
```

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 6]
```diff
@@ -59,6 +59,18 @@ This means it was possible for some projects to compile without any declared dep
...
+However, the configuration cache subsystem was not following that pattern before this release, which could be confusing to users.
```
Comment (lkasso) [inline platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:69]:
```suggestion
However, before this release, the configuration cache subsystem did not follow this pattern, potentially causing confusion.
```

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 7]
```diff
@@ -59,6 +59,18 @@ This means it was possible for some projects to compile without any declared dep
...
+
+In order to address that, in the present release, all code originally under `org.gradle.configurationcache*` packages
```
Comment (lkasso) [inline platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:71]:
```suggestion
To address this issue, in the current release, all code originally under the `org.gradle.configurationcache*` packages
```

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 8]
```diff
@@ -59,6 +59,18 @@ This means it was possible for some projects to compile without any declared dep
...
+(which was never API) has been moved to new internal (`org.gradle.internal.*`) packages.
```
Comment (lkasso) [inline platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:72]:
```suggestion
(which was never API) has been moved to new internal packages (`org.gradle.internal.*`).
```

**Review by lkasso — CHANGES_REQUESTED** [review]

**Review by bamboo — APPROVED** [review]
💯

**Comment by abstratt** [comment]
Thanks, @lkasso, all edits accepted

**Pull request merged**